exchanges/refocus/src/refocusExchange.ts

```diff
@@ -0,0 +1,50 @@
+import { pipe, tap } from 'wonka';
+import { Exchange, Operation } from '@urql/core';
+
+export const refocusExchange = (): Exchange => {
+  return ({ client, forward }) => ops$ => {
+    const watchedOperations = new Map<number, Operation>();
+    const observedOperations = new Map<number, number>();
+    const keys: Array<number> = [];
+
+    window.addEventListener('focus', () => {
+      keys.forEach(key => {
+        const op = watchedOperations.get(key) as Operation;
+        client.reexecuteOperation(
+          client.createRequestOperation(
+            'query',
+            { key: op.key, query: op.query, variables: op.variables },
+            { requestPolicy: 'cache-and-network' }
+          )
+        );
+      });
+    });
+
+    const processIncomingOperation = (op: Operation) => {
+      if (op.operationName !== 'query' && op.operationName !== 'teardown')
+        return;
+
+      if (op.operationName === 'query' && !observedOperations.has(op.key)) {
+        observedOperations.set(op.key, 1);
+        watchedOperations.set(op.key, op);
+        keys.push(op.key);
+      } else if (op.operationName === 'query') {
+        const observedCount = observedOperations.get(op.key) as number;
+        observedOperations.set(op.key, observedCount + 1);
```

Comment:
Shouldn't the count be irrelevant since `teardown` only kicks in when the observing count has already reached zero? Apart from that `reexecuteOperation` should be a nice second protection against this since it's a noop for unobserved queries